src/Analyzer/QueryTreePassManager.cpp

```diff
@@ -61,7 +61,7 @@ namespace ErrorCodes
 namespace
 {
 
-#ifndef NDEBUG
+#if defined(ABORT_ON_LOGICAL_ERROR)
```

Comment:
?

Reply:
[`7c01705` (#59926)](https://github.com/ClickHouse/ClickHouse/pull/59926/commits/7c0170523102e0571213e383415d2e6171f5e5d6)

Reply:
The initial issue had been catched by ValidationChecker, which is enabled for sanitizers builds in this patch, to catch errors for sanitizers as well, since for instance there are no integration tests for debug build